### PR TITLE
🌐 [i18n-KO] Translated `semantic_segmentation.md` to Korean

### PR DESCRIPTION
**Pull request by jungnerd**
# What does this PR do?

Translated the `semantic_segmentation.md` file of the documentation to Korean.
Thank you in advance for your review. 

Part of https://github.com/huggingface/transformers/issues/20179

## Before reviewing
- [x] Check for missing / redundant translations (번역 누락/중복 검사)
- [x] Grammar Check (맞춤법 검사)
- [x] Review or Add new terms to glossary (용어 확인 및 추가)
- [x] Check Inline TOC (e.g. `[[lowercased-header]]`)
- [ ] Check live-preview for gotchas (live-preview로 정상작동 확인)

## Who can review? (Initial)

Team PseudoLab, may you please review this PR? @0525hhgus, @kihoon71, @sim-so, @gabrielwithappy, @HanNayeoniee, @wonhyeongseo

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: docs/source/ko/tasks/semantic_segmentation.md [hunk 1]
```diff
@@ -0,0 +1,591 @@
...
+
+```py
+>>> from huggingface_hub import notebook_login
+
+>>> notebook_login()
+```
+
+## SceneParse150 데이터 세트 불러오기[[load-sceneparse150-dataset]]
+
+🤗 Datasets 라이브러리에서 SceneParse150 데이터 세트의 더 작은 부분 집합을 가져오는 것으로 시작합니다. 이렇게 하면 전체 데이터 세트에 대한 훈련에 많은 시간을 할애하기 전에 실험을 통해 모든 것이 제대로 작동하는지 확인할 수 있습니다.
```
Comment (wonhyeongseo) [inline docs/source/ko/tasks/semantic_segmentation.md:57]:
```suggestion
🤗 Datasets 라이브러리에서 SceneParse150 데이터 세트의 더 작은 부분 집합을 가져오는 것으로 시작합니다. 이렇게 하면 데이터 세트 전체에 대한 훈련에 많은 시간을 할애하기 전에 실험을 통해 모든 것이 제대로 작동하는지 확인할 수 있습니다.
```

앞서 나온 `데이터 세트의 더 작은 부분 집합`과 단어 순서가 똑같도록 `데이터 세트(의) 전체`로 바꿔봤습니다.

File: docs/source/ko/tasks/semantic_segmentation.md [hunk 2]
```diff
@@ -0,0 +1,591 @@
...
+{'image': <PIL.JpegImagePlugin.JpegImageFile image mode=RGB size=512x683 at 0x7F9B0C201F90>,
+ 'annotation': <PIL.PngImagePlugin.PngImageFile image mode=L size=512x683 at 0x7F9B0C201DD0>,
+ 'scene_category': 368}
+```
+
+- `image`: 장면의 PIL 이미지입니다.
+- `annotation`: 분할 지도(segmentation map)의 PIL 이미지입니다. 모델의 타겟이기도 합니다.
+- `scene_category`: "주방" 또는 "사무실"과 같이 이미지 장면을 설명하는 카테고리 ID입니다. 이 가이드에서는 둘 다 PIL 이미지인 `image`와 `annotation`만을 사용합니다.
+
+나중에 모델을 설정할 때 유용하게 사용할 수 있도록 레이블 ID를 레이블 클래스에 매핑하는 사전도 만들고 싶을 것입니다. 허브에서 매핑을 다운로드하고 `id2label` 및 `label2id` 사전을 만드세요:
```
Comment (wonhyeongseo) [inline docs/source/ko/tasks/semantic_segmentation.md:86]:
```suggestion
나중에 모델을 설정할 때 유용하게 사용할 수 있도록 레이블 ID를 레이블 클래스에 매핑하는 사전도 만들고 싶을 것입니다. Hub에서 매핑을 다운로드하고 `id2label` 및 `label2id` 사전을 만드세요:
```

File: docs/source/ko/tasks/semantic_segmentation.md [hunk 3]
```diff
@@ -0,0 +1,591 @@
...
+>>> filename = "ade20k-id2label.json"
+>>> id2label = json.load(open(cached_download(hf_hub_url(repo_id, filename, repo_type="dataset")), "r"))
+>>> id2label = {int(k): v for k, v in id2label.items()}
+>>> label2id = {v: k for k, v in id2label.items()}
+>>> num_labels = len(id2label)
+```
+
+## 전처리하기[[preprocess]
+
+다음 단계는 모델에 사용할 이미지와 주석을 준비하기 위해 SegFormer 이미지 프로세서를 불러오는 것입니다. 우리가 사용하는 데이터 세트와 같은 일부 데이터 세트는 배경 클래스로 제로 인덱스를 사용합니다. 하지만, 배경 클래스는 150개의 클래스에 실제로는 포함되지 않기 때문에 `reduce_labels=True` 를 설정해 모든 레이블에서 배경 클래스를 제거해야 합니다. 제로 인덱스는 `255`로 대체되므로 SegFormer의 손실 함수에서 무시됩니다:
```
Comment (wonhyeongseo) [inline docs/source/ko/tasks/semantic_segmentation.md:102]:
```suggestion
다음 단계는 모델에 사용할 이미지와 주석을 준비하기 위해 SegFormer 이미지 프로세서를 불러오는 것입니다. 우리가 사용하는 데이터 세트와 같은 일부 데이터 세트는 배경 클래스로 제로 인덱스를 사용합니다. 하지만 배경 클래스는 150개의 클래스에 실제로는 포함되지 않기 때문에 `reduce_labels=True` 를 설정해 모든 레이블에서 배경 클래스를 제거해야 합니다. 제로 인덱스는 `255`로 대체되므로 SegFormer의 손실 함수에서 무시됩니다:
```

File: docs/source/ko/tasks/semantic_segmentation.md [hunk 4]
```diff
@@ -175,9 +175,9 @@ Define two separate transformation functions:
 ...     return image
 ```
 
-Next, create two preprocessing functions to prepare batches of images and annotations for the model. These functions apply
-the image transformations and use the earlier loaded `image_processor` to convert the images into `pixel_values` and
-annotations to `labels`. `ImageProcessor` also takes care of resizing and normalizing the images.
+그런 다음 두 개의 전처리 함수를 만들어 모델에 대한 이미지 및 주석 배치를 준비합니다. 이 함수들은
+이미지 변환을 적용하고 이전에 로드한 `image_processor`를 사용하여 이미지를 `pixel_values`로, 주석을 `label`로 변환합니다.
+어노테이션을 `레이블`로 변환합니다. `ImageProcessor` 는 이미지의 크기 조정과 정규화도 처리합니다.
```
Comment (wonhyeongseo) [inline docs/source/ko/tasks/semantic_segmentation.md:180]:
```suggestion
그런 다음 모델을 위해 두 개의 전처리 함수를 만들어 이미지 및 주석 배치를 준비합니다. 이 함수들은 이미지 변환을 적용하고 이전에 로드한 `image_processor`를 사용하여 이미지를 `pixel_values`로, 주석을 `label`로 변환합니다. `ImageProcessor` 는 이미지의 크기 조정과 정규화도 처리합니다.
```
중복된 부분을 없앴어요 :)

File: docs/source/ko/tasks/semantic_segmentation.md [hunk 5]
```diff
@@ -193,9 +193,8 @@ annotations to `labels`. `ImageProcessor` also takes care of resizing and normal
 ...     inputs = image_processor(images, labels)
 ...     return inputs
 ```
-
-To apply the preprocessing transformations over the entire dataset, use the 🤗 Datasets [`~datasets.Dataset.set_transform`] function.
-The transform is applied on the fly which is faster and consumes less disk space:
+전체 데이터 집합에 전처리 변환을 적용하려면 🤗 Datasets [`~datasets.Dataset.set_transform`] 함수를 사용하세요.
+즉시 변환이 적용되기 때문에 더 빠르고 디스크 공간을 덜 차지합니다:
```
Comment (wonhyeongseo) [inline docs/source/ko/tasks/semantic_segmentation.md:197]:
```suggestion

전체 데이터 집합에 전처리 변환을 적용하려면 🤗 Datasets [`~datasets.Dataset.set_transform`] 함수를 사용하세요.
즉시 변환이 적용되기 때문에 더 빠르고 디스크 공간을 덜 차지합니다:
```
앞에 있던 공간을 다시 복구했습니다.

File: docs/source/ko/tasks/semantic_segmentation.md [hunk 6]
```diff
@@ -204,18 +203,16 @@ The transform is applied on the fly which is faster and consumes less disk space
...
 
 ```py
 >>> import evaluate
 
 >>> metric = evaluate.load("mean_iou")
 ```
-
-Then create a function to [`~evaluate.EvaluationModule.compute`] the metrics. Your predictions need to be converted to
-logits first, and then reshaped to match the size of the labels before you can call [`~evaluate.EvaluationModule.compute`]:
+그런 다음 메트릭을 [`~evaluate.EvaluationModule.compute`]하는 함수를 만듭니다. 예측을 먼저 로짓으로 변환한 다음, 레이블의 크기에 맞게 모양을 다시 지정해야 [`~evaluate.EvaluationModule.compute`]를 호출할 수 있습니다:
```
Comment (wonhyeongseo) [inline docs/source/ko/tasks/semantic_segmentation.md:215]:
```suggestion

그런 다음 메트릭을 [`~evaluate.EvaluationModule.compute`]하는 함수를 만듭니다. 예측을 먼저 로짓으로 변환한 다음, 레이블의 크기에 맞게 모양을 다시 지정해야 [`~evaluate.EvaluationModule.compute`]를 호출할 수 있습니다:
```

File: docs/source/ko/tasks/semantic_segmentation.md [hunk 7]
```diff
@@ -287,30 +284,30 @@ logits first, and then reshaped to match the size of the labels before you can c
...
+이제 세 단계만 남았습니다:
 
-1. Define your training hyperparameters in [`TrainingArguments`]. It is important you don't remove unused columns because this'll drop the `image` column. Without the `image` column, you can't create `pixel_values`. Set `remove_unused_columns=False` to prevent this behavior! The only other required parameter is `output_dir` which specifies where to save your model. You'll push this model to the Hub by setting `push_to_hub=True` (you need to be signed in to Hugging Face to upload your model). At the end of each epoch, the [`Trainer`] will evaluate the IoU metric and save the training checkpoint.
-2. Pass the training arguments to [`Trainer`] along with the model, dataset, tokenizer, data collator, and `compute_metrics` function.
-3. Call [`~Trainer.train`] to finetune your model.
+1. 학습 하이퍼파라미터를 [`TrainingArguments`]에 정의합니다. `image` 열이 삭제되기 때문에 사용하지 않는 열을 제거하지 않는 것이 중요합니다. `image` 열이 없으면 `pixel_values`을 생성할 수 없습니다. 이런 경우를 방지하려면 `remove_unused_columns=False`로 설정하세요! 유일하게 필요한 다른 매개변수는 모델을 저장할 위치를 지정하는 `output_dir`입니다. `push_to_hub=True`를 설정하여 이 모델을 허브에 푸시합니다(모델을 업로드하려면 허깅 페이스에 로그인해야 합니다). 각 에포크가 끝날 때마다 [`Trainer`]가 IoU 메트릭을 평가하고 학습 체크포인트를 저장합니다.
```
Comment (wonhyeongseo) [inline docs/source/ko/tasks/semantic_segmentation.md:308]:
```suggestion
1. 학습 하이퍼파라미터를 [`TrainingArguments`]에 정의합니다. `image` 열이 삭제되기 때문에 사용하지 않는 열을 제거하지 않는 것이 중요합니다. `image` 열이 없으면 `pixel_values`을 생성할 수 없습니다. 이런 경우를 방지하려면 `remove_unused_columns=False`로 설정하세요! 유일하게 필요한 다른 매개변수는 모델을 저장할 위치를 지정하는 `output_dir`입니다. `push_to_hub=True`를 설정하여 이 모델을 Hub에 푸시합니다(모델을 업로드하려면 Hugging Face에 로그인해야 합니다). 각 에포크가 끝날 때마다 [`Trainer`]가 IoU 메트릭을 평가하고 학습 체크포인트를 저장합니다.
```

File: docs/source/ko/tasks/semantic_segmentation.md [hunk 8]
```diff
@@ -340,8 +337,7 @@ At this point, only three steps remain:
 
 >>> trainer.train()
 ```
-
-Once training is completed, share your model to the Hub with the [`~transformers.Trainer.push_to_hub`] method so everyone can use your model:
+학습이 완료되면, 모두가 당신의 모델을 사용할 수 있도록 [`~transformers.Trainer.push_to_hub`] 메서드를 사용해 허브에 모델을 공유하세요:
```
Comment (wonhyeongseo) [inline docs/source/ko/tasks/semantic_segmentation.md:340]:
```suggestion
학습이 완료되면, 누구나 모델을 사용할 수 있도록 [`~transformers.Trainer.push_to_hub`] 메서드를 사용해 Hub에 모델을 공유하세요:
```

File: docs/source/ko/tasks/semantic_segmentation.md [hunk 9]
```diff
@@ -436,8 +429,8 @@ and use the [`PushToHubCallback`] to upload the model:
 >>> callbacks = [metric_callback, push_to_hub_callback]
 ```
 
-Finally, you are ready to train your model! Call `fit()` with your training and validation datasets, the number of epochs,
-and your callbacks to fine-tune the model:
+이제 모델을 훈련할 준비가 되었습니다! 훈련 및 검증 데이터 세트, 에포크 수와 함께 `fit()`을 호출하고,
+콜백을 사용하여 모델을 미세 조정합니다:
```
Comment (wonhyeongseo) [inline docs/source/ko/tasks/semantic_segmentation.md:433]:
```suggestion
이제 모델을 훈련할 준비가 되었습니다! 훈련 및 검증 데이터 세트, 에포크 수와 함께 `fit()`을 호출하고, 콜백을 사용하여 모델을 미세 조정합니다:
```

File: docs/source/ko/tasks/semantic_segmentation.md [hunk 10]
```diff
@@ -548,22 +541,22 @@ Load an image processor to preprocess the image and return the input as TensorFl
...
-Pass your input to the model and return the `logits`:
+모델에 입력을 전달하고 `logits`를 반환합니다:
 
 ```py
 >>> from transformers import TFAutoModelForSemanticSegmentation
 
 >>> model = TFAutoModelForSemanticSegmentation.from_pretrained("MariaK/scene_segmentation")
 >>> logits = model(**inputs).logits
 ```
+그런 다음 로그를 원본 이미지 크기로 재조정하고 클래스 차원에 argmax를 적용합니다:
```
Comment (wonhyeongseo) [inline docs/source/ko/tasks/semantic_segmentation.md:552]:
```suggestion

그런 다음 로그를 원본 이미지 크기로 재조정하고 클래스 차원에 argmax를 적용합니다:
```

**Review by wonhyeongseo — CHANGES_REQUESTED** [review]
안녕하세요 우준님, 추석 연휴 잘 보내셨나요? 🌝🐇

간단하게 Hub와 같은 제품명이나 빈줄을 일치시키고, 문맥상 자연스러워 보이게 리뷰를 거쳤습니다. 번역을 계속해주셔서 정말 감사합니다! 💯

File: docs/source/ko/tasks/semantic_segmentation.md [hunk 11]
```diff
@@ -0,0 +1,591 @@
...
+>>> trainer.push_to_hub()
+```
+</pt>
+</frameworkcontent>
+
+<frameworkcontent>
+<tf>
+<Tip>
+
+Keras로 모델을 미세 조정하는 데 익숙하지 않은 경우, 먼저 [기본 튜토리얼](./training#train-a-tensorflow-model-with-keras)을 확인해보세요!
```
Comment (stevhliu) [inline docs/source/ko/tasks/semantic_segmentation.md:352]:
```suggestion
Keras로 모델을 미세 조정하는 데 익숙하지 않은 경우, 먼저 [기본 튜토리얼](../training#train-a-tensorflow-model-with-keras)을 확인해보세요!
```

**Review by stevhliu — APPROVED** [review]
Thanks for translating this guide! 💯
